### DISCUSSION TITLE
Custom join field or how to use RelationshipTable component

### DISCUSSION THREAD

**Discussion by HriBB** [payloadcms/payload#9049] [Q&A] [answered]
I am trying to create a custom `join` field component.

```tsx
import { JoinField } from 'payload'

export const locationSectorsField: JoinField = {
  name: 'sectors',
  type: 'join',
  collection: 'sectors',
  on: 'location',
  defaultSort: 'order',
  admin: {
    components: {
      Field: '@/collections/Locations/fields/LocationSectorsField',
    },
  },
}
```

Component `RelationshipTable` is not exported from `@payloadcms/ui`. If I import it from `@payloadcms/ui/elements/RelationshipTable`, I get this weird `useConfig` error

```tsx
'use client'
import React, { useMemo } from 'react'
import type { JoinFieldClient, JoinFieldClientComponent, PaginatedDocs, Where } from 'payload'
import { fieldBaseClass, FieldLabel } from '@payloadcms/ui'
import { useFieldProps } from '@payloadcms/ui'
import { useField } from '@payloadcms/ui'
import { useDocumentInfo } from '@payloadcms/ui'
//import { RelationshipTable } from '@/components/RelationshipTable'
import { RelationshipTable } from '@payloadcms/ui/elements/RelationshipTable'
import { withCondition } from '@/components/withCondition'

const JoinFieldComponent: JoinFieldClientComponent = function JoinFieldComponent(props) {
  // ...
}
```

![image](https://github.com/user-attachments/assets/16ced231-13bd-453b-8651-9c9e50f876b3)

If I copy the entire `RelationshipTable` into the project and import from there, I don't get the error, but then the Columns logic does not work ...

```tsx
import { RelationshipTable } from '@/components/RelationshipTable'
```

![image](https://github.com/user-attachments/assets/fefa2379-4045-4a9e-8eb7-e7a45d4db2e9)

So the question is, how can we use `RelationshipTable` in our custom components?

**Comment by tyteen4a03** [comment]
If you need the component exported, you can open an issue.

**Reply by revnelson** [reply]
This is related to my issue [here](https://github.com/payloadcms/payload/issues/8822) (3 weeks ago). In my [response](https://github.com/payloadcms/payload/discussions/3960#discussioncomment-11179766) to another discussion thread about this, I also pointed out the need for exported Inputs for each field type, as well as associated helper functions.

**Comment by DanRibbens** [comment, marked as answer]
Here here! https://github.com/payloadcms/payload/pull/10084
